client/src/components/CardDialog/CardDialog.tsx

```diff
@@ -10,38 +10,36 @@ import {
   DialogActions,
   Divider,
 } from '@material-ui/core';
-import cardDialogStyles from './cardDialogStyles';
-import useColorTagStyles from '../Kanban/shared/colorStyles';
-import { useKanban } from '../../context/useKanbanContext';
 import ClearIcon from '@material-ui/icons/Clear';
 import ImportContactsOutlinedIcon from '@material-ui/icons/ImportContactsOutlined';
 import AssignmentOutlinedIcon from '@material-ui/icons/AssignmentOutlined';
 import ScheduleIcon from '@material-ui/icons/Schedule';
 import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline';
+import cardDialogStyles from './cardDialogStyles';
+import useColorTagStyles from '../Kanban/shared/colorStyles';
+import { useKanban } from '../../context/useKanbanContext';
 
-type DialogProps = {
+interface DialogProps {
   name: string;
   tag: string;
   columnId: string;
   id: string;
-};
+}
+
+const AddCardDialog = ({ name = 'blank', columnId, tag = 'white', id }: DialogProps): JSX.Element | null => {
+  const { addCard, activeBoard } = useKanban();
 
-const AddCardDialog = ({ name = 'blank', columnId, tag = 'white', id }: DialogProps): JSX.Element => {
   const [open, setOpen] = useState(false);
   const classes = cardDialogStyles();
   const colorClasses = useColorTagStyles({ tag });
-  const { addCard, columns } = useKanban();
+  const columns = activeBoard.columns;
 
-  const handleClickOpen = () => {
-    setOpen(true);
-  };
+  const handleClickOpen = () => setOpen(true);
 
-  const handleClose = () => {
-    setOpen(false);
-  };
+  const handleClose = () => setOpen(false);
 
   const getColNameById = (columnId: string): string => {
-    const matchingColumns = columns.filter((col) => col.id === columnId);
+    const matchingColumns = columns.filter((col: { _id: string }) => col._id === columnId);
```

Comment:
this may actually conflict with @moffatethan who I think recently changed some of the `_id`s to `id`.
(eg. https://github.com/hatchways/team-rambo/pull/52/files#diff-5058ad396198f940302338e8e76ec588d1b9fcea9467f191d7e1a1bbbf727880R5). So let's pause on this until his PR is in so you don't do too much duplicate work.

Reply:
Yeah, I moved this logic out of the card dialog into the Kanban context.

Reply:
Is your PR ready to merge?

Reply:
Yeah, I pushed the code into main